### DISCUSSION TITLE
Loading 2D DEER Data Sets

### DISCUSSION THREAD

**Discussion by ParkerPBD** [JeschkeLab/DeerLab#373] [Q&A] [answered]
I'm trying to load 2D data and according to the deerload documentation says it should return variables as 

> 2D-datasets: the M time-axes of N points are returned as a two-dimensional ndarray of shape (N,M). The i-th axis can be accessed via t[:,i]

I seem to be having an issue though as I'm running the following 

```
import numpy as np 
import matplotlib.pyplot as plt  
import deerlab as dl

t,Vexp = dl.deerload(filepath)

print(type(Vexp))
print(type(t))
```

And getting

> <class 'numpy.ndarray'>
<class 'list'>

As the output.

The same script returns both variable as ndarrays if I use some of my 1D data. Is there an extra step or change that needs to be made for loading the 2D data that I'm not seeing? I didn't see any guides on loading 2D data.

Thanks so much for the help.

**Comment by mtessmer** [comment, marked as answer]
When loading 2d data sets t returns both the time domain and and index. You can seperate them:

```python
(t, idx), Vexp = dl.deerload(file)
```
